openstack/identity/v3/limits/requests.go

```diff
@@ -5,6 +5,13 @@ import (
 	"github.com/gophercloud/gophercloud/pagination"
 )
 
+// Get retrieves details on a single limit, by ID.
+func GetEnforcementModel(client *gophercloud.ServiceClient) (r EnforcementModelResult) {
+	resp, err := client.Get(enforcementModelURL(client), &r.Body, nil)
+	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)
+	return
+}
+
```

Comment:
Can't we combine the GetEnforcementModel and Extract functions? I don't see the advantage of having them separate as you would always need to do both. This would result in less code and GetEnforcementModel would then return an actual EnforcementModel.
```suggestion
// Get retrieves details on a single limit, by ID.
func GetEnforcementModel(client *gophercloud.ServiceClient) (*EnforcementModel, error) {
	var r gophercloud.Result
	resp, err := client.Get(enforcementModelURL(client), &r.Body, nil)
	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)

	var out struct {
		Model *EnforcementModel `json:"model"`
	}
	err = r.ExtractInto(&out)
	return out.Model, err
}

```

Reply:
It's seems to be a convention in gophercloud I am no keen on breaking.

Reply:
It's as such, because you can cast your response to a different model than in the library (did that in the past) and extract params that would otherwise be lost (some openstack extensions add to these responses, and new versions also have additional keys present)